### PR TITLE
[#5768] improvement(CLI): improve formatted output

### PR DESCRIPTION
**Pull request by Abyss-lord**
### What changes were proposed in this pull request?

Enhanced the Gravitino CLI to support properly formatted output. Fixed alignment issues in formatted output when it includes Chinese characters.
The bounding box length calculation is now based on the number of characters rather than the character display width . The correct output should look like this(Here are the results of my local tests):

image

Only change the implementation of the TableFormatImpl class, the same effect is achieved when the `catalog list --output table` command is supported

### Why are the changes needed?

Fix: #5768

### Does this PR introduce _any_ user-facing change?

NO.

### How was this patch tested?

```shell
gcli catalog details --name <catalog>  --metalake <metalake> --output table
gcli metalake details --name <metalake>  --output table
gcli metalake list --output table
```

### CODE REVIEW

**Comment by Abyss-lord** [comment]
@xunliu   I’m new to this project and would greatly appreciate your help reviewing this PR. Could you please take a look when you have time? Thank you so much for your guidance!  if possible, could you grant me the necessary permissions to run workflows?

**Comment by Abyss-lord** [comment]
@mchades  I’m new to this project and would greatly appreciate your help reviewing this PR. Could you please take a look when you have time? Thank you so much for your guidance! if possible, could you grant me the necessary permissions to run workflows?

**Comment by mchades** [comment]
CI is passed now, we will review this ASAP

File: clients/cli/src/test/java/org/apache/gravitino/cli/integration/test/TableFormatOutputIT.java [hunk 1]
```diff
@@ -152,11 +154,11 @@ public void testCatalogDetailsCommand() {
...
-        "+----------+------------+-----------------+---------+\n"
-            + "| catalog  | type       | provider        | comment |\n"
-            + "+----------+------------+-----------------+---------+\n"
-            + "| postgres | RELATIONAL | jdbc-postgresql | null    |\n"
-            + "+----------+------------+-----------------+---------+",
+        "+----------+------------+-----------------+-------------------+\n"
+            + "| catalog  | type       | provider        | comment           |\n"
+            + "+----------+------------+-----------------+-------------------+\n"
+            + "| postgres | RELATIONAL | jdbc-postgresql | catalog，用于测试 |\n"
+            + "+----------+------------+-----------------+-------------------+",
```
Comment (xunliu) [inline clients/cli/src/test/java/org/apache/gravitino/cli/integration/test/TableFormatOutputIT.java:161]: Don't modify original test case `testCatalogDetailsCommand()`

Please add a new test `postgres2` case to standalone test full corner character
```
testCatalogDetailsCommandFullCornerCharacter() {
    // Create a byte array output stream to capture the output of the command
    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
    PrintStream originalOut = System.out;
    System.setOut(new PrintStream(outputStream));
    String[] args = {
      "catalog",
      "details",
      commandArg(GravitinoOptions.METALAKE),
      "my_metalake",
      commandArg(GravitinoOptions.NAME),
      "postgres2",
      commandArg(GravitinoOptions.OUTPUT),
      "table",
      commandArg(GravitinoOptions.URL),
      gravitinoUrl
    };
    Main.main(args);
...
```
This way, We can test full-corner characters and half-corner characters. :-)

File: clients/cli/src/test/java/org/apache/gravitino/cli/integration/test/TableFormatOutputIT.java [hunk 2]
```diff
@@ -59,7 +59,9 @@ public void startUp() {
       commandArg(GravitinoOptions.PROPERTIES),
       "jdbc-url=jdbc:postgresql://postgresql-host/mydb,jdbc-user=user,jdbc-password=password,jdbc-database=db,jdbc-driver=org.postgresql.Driver",
       commandArg(GravitinoOptions.URL),
-      gravitinoUrl
+      gravitinoUrl,
+      commandArg(GravitinoOptions.COMMENT),
+      "catalog，用于测试"
```
Comment (xunliu) [inline clients/cli/src/test/java/org/apache/gravitino/cli/integration/test/TableFormatOutputIT.java:64]: Don't modify original catalog(`postgres`), and I think we can create a new catalog(`postgres2`) have comments
```
    String[] create_catalog2_args = {
      "catalog",
      "create",
      commandArg(GravitinoOptions.METALAKE),
      "my_metalake",
      commandArg(GravitinoOptions.NAME),
      "postgres2",
      commandArg(GravitinoOptions.PROVIDER),
      "postgres2",
      commandArg(GravitinoOptions.PROPERTIES),
      "jdbc-url=jdbc:postgresql://postgresql-host/mydb,jdbc-user=user,jdbc-password=password,jdbc-database=db,jdbc-driver=org.postgresql.Driver",
      commandArg(GravitinoOptions.URL),
      gravitinoUrl
      gravitinoUrl,
      commandArg(GravitinoOptions.COMMENT),
      "catalog，用于测试"
    };
    Main.main(create_catalog2_args);
```

Comment (Abyss-lord) [reply clients/cli/src/test/java/org/apache/gravitino/cli/integration/test/TableFormatOutputIT.java]: done, plz check it again.

**Comment by xunliu** [comment]
hi @Abyss-lord 
This PR CI failed, Please fix it.

**Comment by Abyss-lord** [comment]
> hi @Abyss-lord This PR CI failed, Please fix it.

done

**Comment by xunliu** [comment]
hi @Abyss-lord 
Can you send your email address to me? I think to discuss with you.
My email: liuxun@apache.org, Tanks!

**Comment by Abyss-lord** [comment]
> hi @Abyss-lord Can you send your email address to me? I think to discuss with you. My email: [liuxun@apache.org](mailto:liuxun@apache.org), Tanks!

hi, @xunliu I just sent you an email.

**Comment by Abyss-lord** [comment]
@xunliu @mchades hello guys, could you plz review this PR again?

**Review by xunliu — APPROVED** [review]
LGTM
